src/System.Reflection.Metadata/src/System/Reflection/Metadata/Internal/Tables.cs

```diff
@@ -2229,53 +2235,52 @@ int containingBlockOffset
             this.Block = containingBlock.GetMemoryBlockAt(containingBlockOffset, (int)(this.RowSize * numberOfRows));
         }
 
-        internal StringHandle GetTypeName(uint rowId)
+        internal StringHandle GetTypeName(int rowId)
         {
-            int rowOffset = (int)(rowId - 1) * this.RowSize;
-            return StringHandle.FromIndex(this.Block.PeekReference(rowOffset + _TypeNameOffset, _IsStringHeapRefSizeSmall));
+            int rowOffset = (rowId - 1) * this.RowSize;
+            return StringHandle.FromOffset(this.Block.PeekHeapReference(rowOffset + _TypeNameOffset, _IsStringHeapRefSizeSmall));
         }
 
-        internal StringHandle GetTypeNamespaceString(uint rowId)
+        internal StringHandle GetTypeNamespaceString(int rowId)
```

Comment:
GetTypeNamespace and GeyTypeNamespaceDefinition for consistency with public API?